### PR TITLE
Bonuses - API calls/routes/DB models

### PR DESCRIPTION
**Pull request by KimarieHB**

### CODE REVIEW

File: database/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+const mongoose = require('mongoose');
+const Soundtrack = require('./models/soundtracks.js');
```
Comment (gektin) [inline database/index.js:2]: I think this would read more clearly if your variable name was also pluralized, but I don't feel too strongly.

File: database/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+mongoose.connect('mongodb://localhost/bonuses', {
+  autoIndex: false,
+  dbName: 'bonuses',
+  useNewUrlParser: true,
+  useUnifiedTopology: true
+}).then( () => {
+  console.log('Connection to MongoDB successful!');
+}).catch( (err) => {
```
Comment (gektin) [inline database/index.js:11]: I don't believe it's best practice to add a padding space after an open parenthesis.

File: database/index.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  console.log('DB connection error:', err);
+});
+
+// Get bonus associated with bundle
+const getBonuses = (bundleId, callback) => {
```
Comment (gektin) [inline database/index.js:16]: It's a bit strange to see you using promises at the top of this file in connecting to your db and then switching here to a callback style. Are you intending to use callbacks throughout your code, or are you planning on using promises? Either way is fine, I believe, but you should pick one and stick with it.

File: database/index.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      callback(err);
+    } else {
+      callback(null, data);
+    }
+  });
+}
+
+// Get individual bonus item info
+const getBonusItem = (title, callback) => {
+  Soundtrack.find({ "bonus_info.title": `${title}` }, (err, data) => {
```
Comment (gektin) [inline database/index.js]: I read this as invoking this `find` function on a key called `bonus_info.title`, does mongoose expect you to enter in key names in double quotes? I believe we're supposed to be using single quotes where possible. I also am not sure that you need the string literal around the value (see your use of `bundleId` without the string literal on line 17).

File: database/index.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      err = new Error(err);
+      callback(err);
+    } else {
+      let song;
+      for (let track of data.bonus_info[0].tracklist) {
+        if (track.name === trackName) {
+          song = {
+            trackNo: track.track_number,
+            name: track.name,
+            songUrl: track.song
```
Comment (gektin) [inline database/index.js]: I believe we are supposed to use trailing commas after the last item in a multi-line object/array.

File: server/index.js [hunk 6]
```diff
@@ -3,6 +3,7 @@ const app = express();
 const morgan = require('morgan');
 const parser = require('body-parser');
 const db = require('../database/index.js');
+
 let port = 3031;
```
Comment (gektin) [inline server/index.js:7]: Do you change your port anywhere? Is it best practice to use a `let` for our ports? This seems like it could be a `const` to me.

File: server/index.js [hunk 7]
```diff
@@ -3,6 +3,7 @@ const app = express();
...
 
 if (port === null || port === '') {
```
Comment (gektin) [inline server/index.js]: I'm confused by this code. Why is this here if you're setting your port above?

File: server/index.js [hunk 8]
```diff
@@ -22,3 +23,42 @@ app.use(parser.json());
...
+      res.json(results);
+    }
+  })
+})
+
+// To render individual bonus upon click selection
+app.get('/bonus-title', (req, res) => {
+  let selectedTitle = req.query.title;
+
+  db.getBonusItem(selectTitle, (err, results) => {
```
Comment (gektin) [inline server/index.js]: I think this should be `selectedTitle` as the first passed in argument here based on the declaration on line 42.

File: server/index.js [hunk 9]
```diff
@@ -22,3 +23,42 @@ app.use(parser.json());
...
+      res.send(err);
+    } else {
+      res.json(results);
+    }
+  })
+})
+
+// To render individual bonus upon click selection
+app.get('/bonus-title', (req, res) => {
+  let selectedTitle = req.query.title;
```
Comment (gektin) [inline server/index.js:38]: Is there a reason to use query params in one case, here on line 42, and req params above, see line 29? If someone else has to hit these routes, then it may be confusing to have one served by one kind of parameter and the other with a different kind.

**Review by gektin — COMMENTED** [review]
I think your code makes sense for the most part. Most of my comments are small style notes or pointing out places where consistency falters. Of those, there's a mix between promises and callbacks that may cause more confusion for you in trying to work on your code, as well as inconsistent use of req params v. query params, which may prove confusing to other people hitting your routes.

I believe you also have a separate PR outstanding with tests for this code, but I just wanted to flag that if you don't, that you'll need to add tests.

File: server/index.js (hunk 6 above)
Comment (KimarieHB) [reply server/index.js:7]: Upon deployment port could change.

File: database/index.js (hunk 2 above)
Comment (KimarieHB) [reply database/index.js:11]: Space removed.

File: database/index.js (hunk 3 above)
Comment (KimarieHB) [reply database/index.js:16]: If there is time after everything is running I will switch my db functions to the wonderful world of promises. For now, I am feeling the necessity of getting some basic, clean working code in my repo. Moving forward I will be more consistent with this.

File: database/index.js (hunk 4 above)
Comment (KimarieHB) [reply database/index.js]: Leftovers from earlier problems. Switched to single quotes and removed string literal.

File: database/index.js (hunk 3 above)
Comment (KimarieHB) [reply database/index.js:16]: Everything, meaning my entire app.

File: database/index.js (hunk 5 above)
Comment (KimarieHB) [reply database/index.js]: Converted. Dangled commas all over the place!

File: server/index.js (hunk 7 above)
Comment (KimarieHB) [reply server/index.js]: Changed.

File: server/index.js (hunk 9 above)
Comment (KimarieHB) [reply server/index.js:38]: Request params do not accommodate multi-word options (at least not the way I was testing it).

File: server/index.js (hunk 8 above)
Comment (KimarieHB) [reply server/index.js]: Changed it to selectedTitle.

**Review by KimarieHB — COMMENTED** [review]
Updated code and will update PR

File: server/index.js (hunk 9 above)
Comment (gektin) [reply server/index.js:38]: camelCase?
